crates/bevy_ecs/src/world/entity_fetch.rs

```diff
@@ -199,22 +211,21 @@ unsafe impl<const N: usize> WorldEntityFetch for &'_ [Entity; N] {
     unsafe fn fetch_mut(
         self,
         cell: UnsafeWorldCell<'_>,
-    ) -> Result<Self::Mut<'_>, EntityFetchError> {
+    ) -> Result<Self::Mut<'_>, EntityMutableFetchError> {
         // Check for duplicate entities.
         for i in 0..self.len() {
             for j in 0..i {
                 if self[i] == self[j] {
-                    return Err(EntityFetchError::AliasedMutability(self[i]));
+                    return Err(EntityMutableFetchError::AliasedMutability(self[i]));
                 }
             }
         }
 
         let mut refs = [const { MaybeUninit::uninit() }; N];
         for (r, &id) in core::iter::zip(&mut refs, self) {
-            let ecell = cell.get_entity(id).ok_or(EntityFetchError::NoSuchEntity(
-                id,
-                cell.entities().entity_does_not_exist_error_details(id),
-            ))?;
+            let ecell = cell
+                .get_entity(id)
+                .map_err(EntityMutableFetchError::EntityDoesNotExist)?;
```

Comment:
If we `impl From<EntityDoesNotExistError> for EntityMutableFetchError`, then `?` will do the error mapping for us, right?  Would that be worth doing?

Reply:
I like that impl: I think it's always the correct mapping.